google/oauth2/credentials.py

```diff
@@ -205,23 +219,31 @@ def refresh(self, request):
 
         scopes = self._scopes if self._scopes is not None else self._default_scopes
 
-        access_token, refresh_token, expiry, grant_response = _client.refresh_grant(
+        (
+            access_token,
+            refresh_token,
+            expiry,
+            grant_response,
+            rapt_token,
+        ) = reauth.refresh_grant(
             request,
             self._token_uri,
             self._refresh_token,
             self._client_id,
             self._client_secret,
-            scopes,
+            scopes=scopes,
+            rapt_token=self._rapt_token,
         )
 
         self.token = access_token
         self.expiry = expiry
         self._refresh_token = refresh_token
         self._id_token = grant_response.get("id_token")
+        self._rapt_token = rapt_token
 
-        if scopes and "scopes" in grant_response:
+        if scopes and "scope" in grant_response:
```

Comment:
Was this wrong earlier? (wondering about the `scopes` -> `scope` change)

Reply:
Yes it is a bug. Response returns `scope` not `scopes`, see step 5 in [google oauth2 doc](https://developers.google.com/identity/protocols/oauth2/web-server) or [oauth2 spec section 5.1](https://tools.ietf.org/html/rfc6749#section-5.1)

The following is the example from the doc:
```
{
  "access_token": "1/fFAGRNJru1FTz70BzhT3Zg",
  "expires_in": 3920,
  "token_type": "Bearer",
  "scope": "https://www.googleapis.com/auth/drive.metadata.readonly",
  "refresh_token": "1//xEoDL4iW3cxlI7yDbSRFYNG01kVKM2C-259HOF2aQbI"
}
```